src/s3/S3Agent.ts

```diff
@@ -58,10 +63,10 @@ class S3Agent {
       throw e;
     }
 
-    const resLink = `${this.s3Controller.getEndpoint()}${resKey}`;
-    const spyLink = `${this.s3Controller.getEndpoint()}${spyKey}`;
-
-    return { resLink, spyLink };
+    return {
+      resLink: `${this.s3Controller.getPublicFileLinkPrefix()}${resKey}`,
```

Comment:
Revisitng my missed comment, let's not expose the `getPublicFileLinkPrefix` guy and make it be returned in the `uploadFile` path.